### ISSUE TITLE
Why BATS syntax raises so many shellsheck errors

### ISSUE THREAD

**Issue by edouard-lopez** [AtomLinter/linter-shellcheck#84]
I wrote the following [`add-channel.tests.bats` tests suite](https://github.com/Coaxis-ASP/opt/blob/02610f9fa69be7265a34ebb7f62a595be6a1e357/daemon/test/add-channel.tests.bats), works fine. However shellsheck keep raising errors.

### Shellcheck errors

All the test case are seen has buggy and `shellcheck` stop on:

     ^-- SC1070: Parsing stopped here because of parsing errors.

**output:**
```
$ shellcheck -s bash ./test/add-channel.tests.bats

In ./test/add-channel.tests.bats line 1:
#!/usr/bin/env bats
^-- SC1008: This shebang was unrecognized. Note that ShellCheck only handles sh/bash/ksh.
                                                                                                                                                                                                                   
In ./test/add-channel.tests.bats line 21:
@test "should raise error without NAME parameter" {                                                                                                                                                                                                                  ^-- SC1083: This { is literal. Check expression (missing ;/\n?) or quote it.
                                                                                                                                                                                                                   
In ./test/add-channel.tests.bats line 25:
    [[ $(remove_ansi ${lines[1]}) == "NAME missing (see 'mast-utils list-host')" ]]
                     ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./test/add-channel.tests.bats line 26:
}
^-- SC1070: Parsing stopped here because of parsing errors.
```

### Question

How do I fix those BATS/shellcheck issues?

**P.S.:** cross-posting [sstephenson/bats#188](https://github.com/sstephenson/bats/issues/188).

**Comment by Arcanemagus** [comment]
This is just a translator of the output of `shellcheck` into a format understood by the Linter system in Atom, it looks like you already found the `shellcheck` issue related to this, please follow there for any updates.
